Product에 Trade List 추가 , Trade 등록 리소스에 최고 구매가와 최저 구매가 포함

kihyuk-jeong opened the pull request:
![피알](https://user-images.githubusercontent.com/39195377/112521227-9f041380-8ddf-11eb-9ea8-a03249792b96.PNG)

1. 상품(Product) 조회시 해당 상품의 입찰 내역을 표시
상품 조회시 상품의 입찰내역을 표시함으로써 한눈에 시세 파악이 가능하도록 구현
- 판매 입찰은 가격을 오름차순으로 전달
- 구매 입찰은 가격을 오름차순으로 전달

2. 상품(Product) 페이지에서 사이즈 선택 후 판매/구매 버튼 클릭시`trades/{productId}?=size`  요청 
- 각 상품별 등록된 Trade 중 즉시구매/즉시판매를 위한 판매입찰최저가 ,구매입찰최고가를 제공

kihyuk-jeong committed:
feat : product에 trade 조회 기능 추가

kihyuk-jeong committed:
[#74] : feat : product -> trade 조회

kihyuk-jeong committed:
[#74] feat : Product에 현재 입찰 목록 추가 외

1. Product에 현재 입찰 목록 추가
2. 입찰 화면에 최고가/ 최저가 표시 (즉시 판매/구매용)

kihyuk-jeong committed:
[#74] refacotr : 메소드 이름 수정

kihyuk-jeong committed:
[#74] refactor : URL 수정

kihyuk-jeong committed:
[#74] chore : 레디스 직렬화 수정 외

1. 직렬화 수정
2. 코딩컨벤션 수정
3. 임시로 비활성화 했던 이메일 인증 발송 로직 주석 해제

kihyuk-jeong committed:
[#74] refactor : TODO 제거 / 공백 제거

f-lab-dev commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,5 @@
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
```
이 파일을 바꿀만한 이유가 있으셨을까요?

ROMANIAPEOPLE replied:
인텔리제이에서 변경 권장 팝업이 뜨길래 변경했는데 정확히 변경된 원인은 잘 모르겠습니다..

f-lab-dev replied:
권장하는데에 있어서는 이유도 분명 같이 말해줄겁니다~ 한번 찾아보셔도 될 것 같아요~

kihyuk-jeong replied:
협업의 경우 개발자마다 사용하는 gradle 버전이 다를 수 있는데 `distributionUrl`을 설정하여 URL을 통해 해당 gradle 버전을 사용할 수 있도록 도와줍니다.
`gradle-6.3-bin.zip` 의 경우 실행파일만 포함하고, `gradle-6.3-all.zip`의 경우 실행파일+document 까지 포함하는데 인텔리제이가  `gradle-6.3-all.zip` 로 설정하도록 권장한 것 같습니다.

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/config/CacheConfig.java:
```diff
@@ -56,16 +57,17 @@ public RedisConnectionFactory redisCacheConnectionFactory() {
         ObjectMapper mapper = new ObjectMapper();
         mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
         mapper.registerModule(new JavaTimeModule());
+        mapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
```
이 옵션은 어떤 역할을 하는 옵션인가요?
jackson 내부에서 어떤 역할을 하는 것인지 알아보시면 좋을 것 같네요~

kihyuk-jeong replied:
공식문서를 찾아본 결과 , 추상클래스/인터페이스또는 객체 유형의 클래스에 대한 다형성 직렬화/역직렬화를 지원합니다. 
추가적으로 `enableDefaultTyping()`은 현재 `@Deprecated` 상태이기 때문에  jackson에서 권장하고 있는`activateDefaultTyping()`으로 변경하였습니다.

kihyuk-jeong committed:
[#74] refactor : objectMapper 수정

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/config/CacheConfig.java:
```diff
@@ -65,9 +66,15 @@ public RedisConnectionFactory redisCacheConnectionFactory() {
         mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
         mapper.registerModule(new JavaTimeModule());
-        mapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
+        mapper.activateDefaultTyping(ptv, ObjectMapper.DefaultTyping.NON_FINAL);
```
이 `DefaultTyping`이 어떻게 동작하는지 더 자세히 알아보시면 좋을 것 같습니다~
생성되는 JSON 문자열이 어떻게 달라지는지 봐도 좋을 것 같고요

kihyuk-jeong replied:
Redis 내부에 어떤 타입을 가진 객체가 캐싱될지 모르기 때문에 역직렬화시 타입을 전부 Object로 처리하게 되는데, 이 경우 배열을 역직렬화 하는 과정에서 오류가 발생하게 됩니다. 따라서 JSON 타입 정보를 명시하여 역직렬화시 타입에 대한 정보를 제공하게 됩니다.

kihyuk-jeong committed:
[#74] jenkinfile 업데이트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
젠킨스 테스트

kihyuk-jeong committed:
jenkinfile 변경

kihyuk-jeong commented:
Jenkins 관련 issue를 새로 파서 진행했어야 하는데 정신없이 하다보니 전혀 관련없는 PR에 진행하게 되었습니다.
앞으로 신경써서 관리하겠습니다!

kihyuk-jeong committed:
판매 / 구매 등록 구현 [즉시 구매] (#83)

* [#76] feat : 즉시 구매 / 즉시 판매 비즈니스 로직 구현

1. 즉시 판매 비즈니스 로직 구현
2. 즉시 구매 비즈니스 로직 구현

 #76 판매 / 구매 등록 구현 [즉시 구매]

* [#76] refactor : 리뷰 반영

1. 자기 자신이 등록한 거래에 대해 즉시구매/즉시판매 불가능 하도록 수정
2. 거래 등록 페이지에서 필요한 리소스 얻어오는 메소드 네임 수정

* [#76] refactor : 주석 제거

* [#76] refactor : 리뷰 반영

1. 카멜케이스 지키도록 수정

kihyuk-jeong committed:
Merge branch 'feature/73' into feature/74
